clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/artifact/KubernetesArtifactConverter.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.kubernetes.v2.artifact;
+
+import com.netflix.spinnaker.clouddriver.kubernetes.KubernetesCloudProvider;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.KubernetesApiVersion;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.KubernetesCoordinates;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.KubernetesKind;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.KubernetesManifest;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+
+public abstract class KubernetesArtifactConverter {
+  abstract public Artifact toArtifact(KubernetesManifest manifest);
+  abstract public KubernetesCoordinates toCoordinates(Artifact artifact);
+  abstract public String getDeployedName(Artifact artifact);
+
+  protected String getType(KubernetesManifest manifest) {
+    return String.join("/",
+        KubernetesCloudProvider.getID(),
+        manifest.getApiVersion().toString() + ":" + manifest.getKind().toString()
+    );
+  }
+
+  private String[] getLatterType(Artifact artifact) {
```

Comment:
This contains enough logic that a unit test would be helpful.

Reply:
done